assets/js/modules/analytics/components/setup/SetupMain.js

```diff
@@ -57,6 +61,36 @@ export default function SetupMain( { finishSetup } ) {
 		select( MODULES_ANALYTICS ).getSetupFlowMode()
 	);
 
+	const { setAccountID } = useDispatch( MODULES_ANALYTICS );
+	const { selectProperty } = useDispatch( MODULES_ANALYTICS_4 );
+	const [ isMatchedAccount, setIsMatchedAccount ] = useState( false );
+	const { findMatchedAccount } = useDispatch( MODULES_ANALYTICS_4 );
+	useEffect( () => {
+		if ( ! accounts ) {
+			return;
+		}
+
+		const fetchMatchedAccount = async () => {
+			setIsMatchedAccount( true );
+			const matchedAccount = await findMatchedAccount();
+			setIsMatchedAccount( false );
+			if ( matchedAccount ) {
+				setAccountID( matchedAccount._accountID );
+				selectProperty( matchedAccount._id );
```

Comment:
We should call the `matchAndSelectProperty` action after you update the `findMatchedAccount` action to return actual account instead of a property.

Reply:
Updated ✅